docs/user-guide/templates/pipeline-templates.md

```diff
@@ -0,0 +1,194 @@
+
+Most templates will tag the most recent version as `latest`, and many templates use either automated testing or manual curation to identify some version as `stable`. These are floating tags, so using them in a job means its template-provided steps may suddenly change.
+
+If no template version is specified, the most recently published will be used. This is usually synonymous with specifying the `latest` tag. It is generally better to explicitly state a template version than to implicitly use `latest`.
+
+The most reliable way to avoid unexpected template changes is to refer to a specific version of the template. For instance, `nodejs/test@1.0.4` is an immutable reference to a particular list of steps. Using a reference such as `nodejs/test@1.0` means that a job will automatically use `nodejs/test@1.0.5` when it becomes available, but that comes with risk of an unexpected change in behavior.
+
+## Version/Tag Semantics
+
+Template versions can be referenced in a variety of ways that express users' trade-off between an unchanging set of steps and automatically using improvements that a template's maintainers have added. As above, these examples reference [nodejs/test template](https://cd.screwdriver.cd/templates/nodejs/test).
+
+* `nodejs/test@latest` - this will use the most recently published version of the template, which may include backwards-incompatible changes, major version changes, etc. The `latest` tag should primarily be used by a template's maintainers and may be unsuitable for production or similarly important builds.
+* `nodejs/test@stable` - this will use a version of the template that its maintainers have designated as sufficiently stable for general usage. It may represent a significant change in capability from older uses of the `stable` tag. Template maintainers should communicate to users when changes to the `stable` tag are not backwards compatible.
+* `nodejs/test@1` - this will use the most recent version of `nodejs/test` that is less than 2.0.0. This is essentially the `latest` tag without crossing a major version boundary.
+* `nodejs/test@1.0` - this will use the most recent version of `nodejs/test` than is less than 1.1.0. This is essentially the `latest` tag without crossing a minor version boundary.
+* `nodejs/test@1.0.4` - this is the most predictable way to specify a pipeline's behavior and is not affected by future changes to the template.
+
+#### Example
+
+For this configuration:
+```yaml
+template: nodejs/test@stable
+shared: 
+  environment:
+    FOO: bar
+```
+
+
+Screwdriver takes the template configuration and plugs it in, so that the `screwdriver.yaml` becomes:
+
+```yaml
+shared:
+  environment:
+    FOO: bar
+jobs:
+    main:
+      image: node:18
+      requires: [~pr, ~commit]
+      steps:
+        - install: npm install
+        - test: npm test
+      secrets:
+            - NPM_TOKEN
+```
```

Comment:
this also looks like a job template, should this be replaced with pipeline template?

Reply:
not sure what you mean by looks like job template?

Reply:
please scrap it, it was me not understanding the config initially, but it is cleared out now.